website/docs/developer/development-guide/code/integration-tests.md

```diff
@@ -0,0 +1,163 @@
+     ::: warning
+     If the catalog is not reset prior to this test, or other tests run in parallel, other errors may be present.
+     Make sure none of these errors contain your randomized keys.
+     :::
+- [Security](#security-paths):
+  1. Follow the Happy test up to contacting the API.
+  1. No internal reconciliation endpoints should be publicly accessible, even if the job-id is known.
+- [Error](#error-paths):
+  1. Requests for a non-existent job and its reports should return HTTP Status 404.
+
+### Ingest
+- [Happy](#happy-path):
+  1. Add files to an S3 bucket that is registered with Orca.
+     Structure should imitate two granules.
+     :::tip
+     Include a 200 GB file to make sure timeouts do not prevent ingest.
+     :::
+     :::tip
+     Test ingesting from Glacier buckets as well as regular buckets.
+     :::
+  1. Call the OrcaCopyToGlacierWorkflow to ingest the granules to Orca.
+     :::tip
+     Make sure to cover ExcludeDataTypes being set, being unset, and excluding files. May require additional tests.
+     :::
+     :::tip
+     Future work will allow us to target multiple buckets with ingest.
+     :::
+  1. Retry calls to the [Catalog API](../../../developer/api/api-gateway.md/#catalog-reporting-api) until entries are found.
+  1. Verify that the files are present in the proper Orca bucket.
+- [Security](#security-paths):
+  1. Follow the Happy test up to calling the workflow.
+  1. Workflow should not be publicly accessible, even if files are valid.
+  1. Follow the Happy test up to contacting the API.
+  1. No catalog endpoints should be publicly accessible, even if the granule-id or other values are known.
+- [Error](#error-paths):
+  1. Requests for catalog info on a non-existent granule/file should return HTTP Status 404.
+
+### Recovery
+- [Happy](#happy-path):
+  1. Ingest randomized data to Orca.
+     :::tip
+     Include a 200 GB file to make sure timeouts do not prevent recovery.
+     :::
+  1. Call the OrcaRecoveryWorkflow to restore the files from Orca to another bucket.
+  1. Retry calls to the [Recovery Granules API](../../../developer/api/api-gateway.md/#recovery-granules-api) until entries are found, and status is `complete`.
+     :::warning
+     Recovery may take up to 4 hours.
+     :::
+  1. Verify that the files are present in the proper Orca bucket.
+- [Security](#security-paths):
+  1. Follow the Happy test up to calling the workflow.
+  1. Workflow should not be publicly accessible, even if files are valid.
+  1. Follow the Happy test up to calling the workflow.
+  1. Workflow should not be able to restore files to arbitrary buckets.
+  1. Follow the Happy test up to contacting the API.
+  1. No recovery endpoints should be publicly accessible, even if the granule-id or other values are known.
+- [Error](#error-paths):
+  1. Requests for recovery info on files that are not being recovered should return HTTP Status 404.
+
```

Comment:
Could also test the following:
- Trying to recover a file that is not in the archive because it has been filtered (*.xml for example)
   - Shows recovery respects filter and does not try to recover it.
- Trying to recover a file that does not exist in the archive at all
   - Validates that file is marked as failed with reason

Reply:
Former added to happy path in tip.
Later added.